3rdparty/CMakeLists.txt

```diff
@@ -1,17 +1,5 @@
-list (APPEND thirdparty_src)
-list (APPEND thirdparty_lib)
-list (APPEND thirdparty_inc_public)
-list (APPEND thirdparty_inc)
-list (APPEND thirdparty_def)
-
 execute_process(COMMAND ${MBEDTLS_PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/config.py -f ${CMAKE_CURRENT_SOURCE_DIR}/../include/mbedtls/config.h get MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED RESULT_VARIABLE result)
```

Comment:
As this is not a project, i suggest turning this file into a script. For this
1.  the file should be renamed to i.e. `3rdparty.cmake`
2. any CMakeLists.txt file to be changed to use `include(${MBEDTLS_DIR}/3rdparrty.cmake)`.

Reply:
This is related to #4270

Reply:
Could you clarify? I don't think `CMakeLists.txt` is intended to represent only root projects - the scheme used by the [official CMake repository](https://github.com/Kitware/CMake/blob/master/Source/CMakeLists.txt) suggests as much, where subcomponents have their own `CMakeLists.txt`. In this case we want to mirror the source directory hierarchy.

Reply:
I answered this here: https://github.com/ARMmbed/mbedtls/issues/4270#issuecomment-810441802

Reply:
I don't think it doesn't make much sense for me to introduce this scheme with this unrelated PR - particularly when it's as controversial as it is - so I'll maintain the status quo and mark these conversations as resolved so that the discussion can be continued on the dedicated issue.

Reply:
Ok.